ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/BlsToExecutionOperationPoolTest.java

```diff
@@ -0,0 +1,265 @@
+  }
+
+  @Test
+  void shouldAddEntryToPool() {
+    when(validator.validateForGossip(any()))
+        .thenReturn(SafeFuture.completedFuture(InternalValidationResult.ACCEPT));
+    final SignedBlsToExecutionChange item = dataStructureUtil.randomSignedBlsToExecutionChange();
+    final SafeFuture<?> future = pool.addLocal(item);
+    assertThat(future.isCompletedNormally()).isTrue();
+    assertThat(pool.size()).isEqualTo(1);
+    assertThat(pool.getAll()).containsExactly(item);
+  }
+
+  @Test
+  void shouldRejectDuplicateEntryFromPool() throws ExecutionException, InterruptedException {
+    final SignedBlsToExecutionChange item = initPoolWithSingleItem();
+
+    final SafeFuture<InternalValidationResult> future2 = pool.addLocal(item);
+    assertThat(future2.get().code()).isEqualTo(InternalValidationResult.IGNORE.code());
+  }
+
+  @Test
+  void shouldRemoveEntryFromPool() {
+    final SignedBlsToExecutionChange item = initPoolWithSingleItem();
+    pool.removeAll(
+        blsToExecutionSchemaSupplier
+            .apply(UInt64.ZERO)
+            .createFromElements(
+                List.of(item, dataStructureUtil.randomSignedBlsToExecutionChange(), item)));
+    assertThat(pool.size()).isEqualTo(0);
+  }
+
+  @Test
+  void shouldAddEntriesToPool() {
+    final SignedBlsToExecutionChange item = initPoolWithSingleItem();
+    final SignedBlsToExecutionChange item2 = dataStructureUtil.randomSignedBlsToExecutionChange();
+    pool.addAll(
+        blsToExecutionSchemaSupplier.apply(UInt64.ZERO).createFromElements(List.of(item, item2)));
+    assertThat(pool.size()).isEqualTo(2);
+    assertThat(pool.getAll()).containsExactlyInAnyOrder(item, item2);
+  }
+
+  private SignedBlsToExecutionChange initPoolWithSingleItem() {
+    when(validator.validateForGossip(any()))
+        .thenReturn(SafeFuture.completedFuture(InternalValidationResult.ACCEPT));
+    final SignedBlsToExecutionChange item = dataStructureUtil.randomSignedBlsToExecutionChange();
+    final SafeFuture<InternalValidationResult> future = pool.addLocal(item);
+    assertThat(future.isCompletedNormally()).isTrue();
+    assertThat(pool.size()).isEqualTo(1);
+    return item;
+  }
+
+  @Test
+  void shouldAddMaxItemsToPool() {
+    final SignedBlsToExecutionChange item = initPoolWithSingleItem();
+    when(validator.validateForGossip(any())).thenReturn(completedFuture(ACCEPT));
+    when(validator.validateForBlockInclusion(any(), any())).thenReturn(Optional.empty());
+    final int maxBlsToExecutionChanges =
+        spec.getGenesisSpecConfig().toVersionCapella().orElseThrow().getMaxBlsToExecutionChanges();
+    while (pool.size() < maxBlsToExecutionChanges) {
+      assertThat(pool.addLocal(dataStructureUtil.randomSignedBlsToExecutionChange())).isCompleted();
+    }
+
+    assertThat(pool.getItemsForBlock(state)).hasSize(maxBlsToExecutionChanges);
+    assertThat(pool.size()).isEqualTo(maxBlsToExecutionChanges);
+    assertThat(pool.getItemsForBlock(state)).contains(item);
+  }
```

Comment:
I am confused by this test case. Are we testing that we can fill the pool or that `getItemsForBlock()` respects the maximum number of operations in the block?

The reason I am confused is that the pool size is defined by `DEFAULT_OPERATION_POOL_SIZE` but in the test case we are using `spec.getGenesisSpecConfig().toVersionCapella().orElseThrow().getMaxBlsToExecutionChanges()`.

It looks like we want to test `getItemsForBlock()`. Maybe it is the name of the test that is misleading?